### PR TITLE
feat: implement election-post

### PR DESCRIPTION
**Pull request by dignifiedquire**
Depends on

- [x] https://github.com/filecoin-project/rust-fil-proofs/pull/948
- [x] https://github.com/filecoin-project/rust-fil-proofs-ffi/pull/36
- [x] https://github.com/filecoin-project/rust-fil-sector-builder/pull/88

Tasks 

- [x] pull in new code
- [x] update code
- [x] update tests

### CODE REVIEW

File: transforms.go [hunk 1]
```diff
@@ -97,6 +117,29 @@ func goSealTicket(src C.sector_builder_ffi_FFISealTicket) SealTicket {
 	}
 }
 
+func goCandidates(src *C.sector_builder_ffi_FFICandidate, size C.size_t) ([]Candidate, error) {
+	candidates := make([]Candidate, size)
+	if src == nil || size == 0 {
+		return candidates, nil
+	}
+
+	ptrs := (*[1 << 30]C.sector_builder_ffi_FFICandidate)(unsafe.Pointer(src))[:size:size]
```
Comment (whyrusleeping) [inline transforms.go:126]: why are we casting this pointer as a gigabyte sized fixed array?

Comment (dignifiedquire) [reply transforms.go]: Not sure tbh, I copied this from the other casts. cc @laser

Comment (dignifiedquire) [reply transforms.go:126]: Not sure tbh, I copied this from the other casts. cc @laser

Comment (laser) [reply transforms.go:126]: See the [CGO Wiki](https://github.com/golang/go/wiki/cgo#turning-c-arrays-into-go-slices) for an explanation, if ye so desire.

File: bindings_test.go [hunk 2]
```diff
@@ -179,11 +179,11 @@ func TestSectorBuilderLifecycle(t *testing.T) {
 	})
 
 	// generate a PoSt
-	proofs, err := sb.GeneratePoSt(ptr, sectorInfo, [32]byte{}, []uint64{})
+	proofs, err := sb.GeneratePoSt(ptr, sectorInfo, [32]byte{}, []sb.Candidate{})
```
Comment (laser) [inline bindings_test.go]: @dignifiedquire - We should exercise the `GenerateCandidates` call in this test. I can update this PR if desired.

**Review by laser — APPROVED** [review]

**Pull request merged**